src/ekat/ekat_pack_kokkos.hpp

```diff
@@ -253,7 +254,10 @@ struct HTDVectorT
 
 template<>
 struct HTDVectorT<bool>
-{ using type = char; };
+{
+  static_assert(sizeof(bool) == sizeof(char));
```

Comment:
Since static assert supports a msg, let's put an error explaining what's happening. To be quick, we can just put the PR number as a reference, since we have some discussion on the reason for this check.